Reporting/ad-extensions-reporter.js

```diff
@@ -0,0 +1,384 @@
+        } else if (includeExtensionsWithZeroImpressions === false) {
+            var impressionsCondition = 'WHERE Impressions > 0 ';
+        } else {
+            throw ("Value of options variable includeExtensionsWithZeroImpressions must be a boolean, either true or false. " +
+                "Currently, it is: " + includeExtensionsWithZeroImpressions)
+        }
+    }
+
+    var placeholderFeedItemReport = AdsApp.report(
+        'SELECT PlaceholderType, AttributeValues, DisapprovalShortNames ' +
+        'FROM PLACEHOLDER_FEED_ITEM_REPORT ' +
+        impressionsCondition +
+        'DURING TODAY'
+    );
+    var reportRows = placeholderFeedItemReport.rows();
+
+    // Process report data, to prepare for outputting to the spreadsheet
+    Logger.log("Processing report data");
+
+    var extensionsByType = {};
+    while (reportRows.hasNext()) {
+        var row = reportRows.next();
+
+        var placeholderTypeId = row['PlaceholderType']
+        if (!(placeholderTypeId in extensionTypeToPlaceholderTables)) {
+            continue;
+        }
+        if (!(placeholderTypeId in extensionsByType)) {
+            extensionsByType[placeholderTypeId] = {};
+        }
+
+        var jsonAttributes = row['AttributeValues'];
+        var jsonDisapprovalNames = row['DisapprovalShortNames'];
+        if ('' === jsonDisapprovalNames.toString()) {
+            var disapprovalNames = [];
+        } else {
+            var disapprovalNames = JSON.parse(jsonDisapprovalNames);
+        };
+        if (!(jsonAttributes in extensionsByType[placeholderTypeId])) {
+            extensionsByType[placeholderTypeId][jsonAttributes] = {};
+            extensionsByType[placeholderTypeId][jsonAttributes]['Disapprovals'] = disapprovalNames;
+            extensionsByType[placeholderTypeId][jsonAttributes]['FeedItemCount'] = 1;
+        } else {
+            extensionsByType[placeholderTypeId][jsonAttributes]['FeedItemCount'] += 1;
+            for (var i = 0, n = disapprovalNames.length; i < n; i++) {
+                var disapprovalName = disapprovalNames[i];
+                var previouslyFoundDisapprovalNames = extensionsByType[placeholderTypeId][jsonAttributes]['Disapprovals'];
+                if (previouslyFoundDisapprovalNames.indexOf(disapprovalName) === -1) {
+                    extensionsByType[placeholderTypeId][jsonAttributes]['Disapprovals'] =
+                        previouslyFoundDisapprovalNames.concat(disapprovalName);
+                }
+            }
+        }
+    }
+
+    // Create a spreadsheet tab for each extension type, and output its data
+    Logger.log("Outputting data to spreadsheet");
+    for (var placeholderTypeId in extensionsByType) {
+        var placeholderTable = extensionTypeToPlaceholderTables[placeholderTypeId];
+        var orderedAttributeIds = Object.keys(placeholderTable).sort();
+        var nameIndex = orderedAttributeIds.indexOf('name');
+        orderedAttributeIds.splice(nameIndex, 1); // Remove 'name'
```

Comment:
function?

Reply:
Yes! It's nice that then we don't need the comment explaining what we're doing, because the function name documents it.